src/cli/export.rs

```diff
@@ -84,12 +84,10 @@ pub async fn init(
 		config,
 	}: ExportCommandArguments,
 ) -> Result<(), Error> {
+	let is_local = any::__into_endpoint(&endpoint)?.parse_kind()?.is_local();
```

Comment:
Out of curiosity, why do we care that `IntoEndpoint` was part of the public API? Seems like a reasonable thing to have at first glance.

Having an undocumented `__into_endpoint` is still part of our public API and your IDE's autocomplete will show it as an option.

Reply:
> Out of curiosity, why do we care that IntoEndpoint was part of the public API?

We want to keep our API surface small. A lot of those deprecations were so we could seal traits like `IntoEndpoint`, which this PR now does. Those were never meant to be used directly. Exposing them leads to [issues like this one](https://github.com/surrealdb/surrealdb/issues/5632).

> Having an undocumented __into_endpoint is still part of our public API and your IDE's autocomplete will show it as an option.

I know. I added a double underscore prefix here as a hint that it's undocumented. I also plan to add a `#[diagnostic::do_not_recommend]` attribute to it and other undocumented API in a follow up PR to switch to Rust 2024. The long term plan is to get rid of undocumented APIs.

Reply:
Ok, sounds good. Thanks for the context!